### ISSUE TITLE
Explicitly require relay-runtime and make it a peerDep

### ISSUE THREAD

**Issue by mattkrick** [facebook/relay#2307]
Today, the docs tell us to install 3 things: babel-plugin-relay, relay-compiler, and react-relay.

I propose we add relay-runtime to the list & make it a peerDep for relay-compiler & react-relay.

This solves 2 problems:
- `import {Environment} from 'relay-runtime'` won't throw errors in IDEs because relay-runtime doesn't exist in your package.json. Telling folks to import from a package that they didn't install makes beginners like me super confused.
- relay-compiler & react-relay both require relay-runtime & each installs a separate version themselves, for a total of 3 versions in your app! If you don't know this, developing locally can be a huge pain & discourage folks from making PRs (personally took me way too longer to figure out the relay-runtime i was editing wasn't the one being used by the compiler).

Happy to make a PR if you agree!

**Comment by alloy** [comment]
I agree that explicitly depending on it if we’re going to import from the package makes sense to me.

I’m a little confused how you end up with all other packages installing their own Relay-runtime, though. Are these different versions? And what package manager (including version) are you using?

**Comment by mattkrick** [comment]
> I’m a little confused how you end up with all other packages installing their own Relay-runtime

This is for local dev. Let's say i'm gonna make a PR to relay-runtime that changes an export from `foo` to `bar`. I have go into relay-compiler & react-relay to symlink them to the one I'm working on, otherwise they'll still use `foo`.
